include/flashinfer/attention/scheduler.cuh

```diff
@@ -529,11 +525,14 @@ inline auto PrefillSplitQOKVIndptr(IdType* qo_indptr_h, IdType* kv_indptr_h,
     o_indptr.push_back(o_indptr.back() + qo_len * num_tiles_kv);
   }
 
+  size_t padded_batch_size =
+      enable_cuda_graph ? std::max(max_batch_size_if_split, total_num_tiles_q) : new_batch_size;
```

Comment:
I suppose `total_num_tiles_q` might still varies even if we fix `total_num_rows` and `batch_size`?

For example, if batch_size=2 and total_num_rows=128, the following two query length distribution should have different `total_num_tiles_q`:

* qo_len[0] = 63, qo_len[1] = 65
* qo_len[0] = 64, qo_len[1] = 64

In the first case, `total_num_tiles_q=ceil_div(63, 64) + ceil_div(65, 64) = 3`, and in the second case `total_num_tiles_q=ceil_div(64, 64) + ceil_div(64, 64) = 2`. When this value is greater than maximum batch size, then padded batch size would change, which changes the pointer offset accordingly and might violate the cudagraph static input requirement.

Reply:
That makes perfect sense, thank you for the clarification!